e2e/tests/visual-regression.spec.ts

```diff
@@ -24,10 +25,38 @@ test.describe('@visual-regression', () => {
         test('matches the saved screenshot', async ({ page }) => {
           const locator = selector ? page.locator(selector) : page.getByRole('document')
 
-          await expect(locator).toHaveScreenshot([
-            screenshotName,
-            'js-disabled.png',
-          ], { scale: 'css', timeout: 20_000 })
+          const boundingBox = selector ? await locator.boundingBox() : null
+
+          if (boundingBox && boundingBox.height > vMax) {
+            const remainder = boundingBox.height % vMax ? [boundingBox.height % vMax] : []
+
+            const batches = [
+              ...Array.from<number>({ length: Math.floor(boundingBox.height / vMax) }).fill(vMax),
+              ...remainder,
+            ]
+
+            const result = batches.map((height, index, arr) =>
+              expect(page).toHaveScreenshot([
+                screenshotName,
+                `js-disabled-${index + 1}-of-${arr.length}.png`,
+              ], {
+                scale: 'css',
+                timeout: 20_000,
+                clip: {
+                  ...boundingBox,
+                  y: boundingBox.y + (vMax * index),
+                  height,
+                },
+                fullPage: true,
+              })
+            )
+            await Promise.all(result)
+          } else {
+            await expect(locator).toHaveScreenshot([
+              screenshotName,
+              'js-disabled.png',
+            ], { scale: 'css', timeout: 20_000 })
+          }
```

Comment:
It seems there is a 32767 pixel limit for screenshots so this will split the screenshots by 30,000px and append the remainder (if above 0) to the end of the array.